### PR TITLE
edit inplace_updates section in docs and update all code blocks

### PR DESCRIPTION
**Pull request by paulaehab**

### CODE REVIEW

File: docs/overview/deep_dive/inplace_updates.rst [hunk 1]
```diff
@@ -287,31 +287,35 @@ The implementations of :func:`ivy.tan` for each backend are as follows.
...
 
+
     tan.support_native_out = True
 
 **TensorFlow** (no :code:`support_native_out` attribute):
 
 .. code-block:: python
 
     def tan(
-        x: Union[tf.Tensor, tf.Variable],
```
Comment (VedPatwardhan) [inline docs/overview/deep_dive/inplace_updates.rst:301]: Could you please revert the change to indentation for this code block?

File: docs/overview/deep_dive/inplace_updates.rst [hunk 2]
```diff
@@ -287,31 +287,35 @@ The implementations of :func:`ivy.tan` for each backend are as follows.
...
+    out: Optional[Union[tf.Tensor, tf.Variable]] = None,
     ) -> Union[tf.Tensor, tf.Variable]:
-        return tf.tan(x)
+    return tf.tan(x)
 
 **PyTorch** (includes :code:`support_native_out` attribute):
 
 .. code-block:: python
 
+
```
Comment (VedPatwardhan) [inline docs/overview/deep_dive/inplace_updates.rst:313]: could you please remove this additional line?

File: docs/overview/deep_dive/inplace_updates.rst [hunk 3]
```diff
@@ -328,7 +332,7 @@ Take a function which has multiple possible "paths" through the code:
 
 .. code-block:: python
 
-    def cholesky(
+        def cholesky(
```
Comment (VedPatwardhan) [inline docs/overview/deep_dive/inplace_updates.rst:335]: another issue with indentation

File: docs/overview/deep_dive/inplace_updates.rst [hunk 4]
```diff
@@ -356,6 +360,7 @@ Another case where we need to use :func:`ivy.inplace_update`_ with a function th
 
 .. code-block:: python
 
+
```
Comment (VedPatwardhan) [inline docs/overview/deep_dive/inplace_updates.rst:363]: additional line

File: docs/overview/deep_dive/inplace_updates.rst [hunk 5]
```diff
@@ -399,20 +407,23 @@ The second and third points are the most important points.
...
 
     def cross_entropy(
-        true: Union[ivy.Array, ivy.NativeArray],
-        pred: Union[ivy.Array, ivy.NativeArray],
-        /,
-        *,
-        axis: Optional[int] = -1,
-        epsilon: float =1e-7,
-        out: Optional[ivy.Array] = None
+    true: Union[ivy.Array, ivy.NativeArray],
```
Comment (VedPatwardhan) [inline docs/overview/deep_dive/inplace_updates.rst:414]: indentation

File: docs/overview/deep_dive/inplace_updates.rst [hunk 6]
```diff
@@ -399,20 +407,23 @@ The second and third points are the most important points.
...
+    axis: int = -1,
+    epsilon: float = 1e-7,
+    reduction: str = "sum",
+    out: Optional[ivy.Array] = None,
     ) -> ivy.Array:
+        ivy.utils.assertions.check_elem_in_list(reduction, ["none", "sum", "mean"])
         pred = ivy.clip(pred, epsilon, 1 - epsilon)
         log_pred = ivy.log(pred)
-        return ivy.negative(ivy.sum(log_pred * true, axis, out=out), out=out)
+        return _reduce_loss(reduction, log_pred * true, axis, out)
```
Comment (VedPatwardhan) [inline docs/overview/deep_dive/inplace_updates.rst:426]: let's pass `out` as a keyword argument following convention

File: docs/overview/deep_dive/inplace_updates.rst [hunk 7]
```diff
@@ -45,7 +45,7 @@ While :class:`ivy.Array` instances will always be inplace updated consistently,
 
 .. code-block:: python
 
-    def inplace_update(
+        def inplace_update(
```
Comment (VedPatwardhan) [inline docs/overview/deep_dive/inplace_updates.rst:48]: indentation

File: docs/overview/deep_dive/inplace_updates.rst [hunk 8]
```diff
@@ -105,7 +105,7 @@ JAX functions also never returns views so additional logic is added to functiona
...
```
Comment (VedPatwardhan) [inline docs/overview/deep_dive/inplace_updates.rst:108]: indentation

File: docs/overview/deep_dive/inplace_updates.rst [hunk 9]
```diff
@@ -146,7 +146,7 @@ Following this, an inplace update is then also performed on the :class:`ivy.Arra
...
```
Comment (VedPatwardhan) [inline docs/overview/deep_dive/inplace_updates.rst:149]: indentation

File: docs/overview/deep_dive/inplace_updates.rst [hunk 10]
```diff
@@ -212,7 +212,7 @@ TensorFlow functions also never returns views so additional logic is added to fu
...
```
Comment (VedPatwardhan) [inline docs/overview/deep_dive/inplace_updates.rst:215]: indentation

**Review by VedPatwardhan — CHANGES_REQUESTED** [review]
Hey @paulaehab, thanks for updating the docs, just requested a few changes 🙂

File: docs/overview/deep_dive/inplace_updates.rst (hunk 7 above)
Comment (paulaehab) [reply docs/overview/deep_dive/inplace_updates.rst:48]: done

File: docs/overview/deep_dive/inplace_updates.rst (hunk 8 above)
Comment (paulaehab) [reply docs/overview/deep_dive/inplace_updates.rst:108]: done

File: docs/overview/deep_dive/inplace_updates.rst (hunk 9 above)
Comment (paulaehab) [reply docs/overview/deep_dive/inplace_updates.rst:149]: done

File: docs/overview/deep_dive/inplace_updates.rst (hunk 10 above)
Comment (paulaehab) [reply docs/overview/deep_dive/inplace_updates.rst:215]: done

File: docs/overview/deep_dive/inplace_updates.rst (hunk 2 above)
Comment (paulaehab) [reply docs/overview/deep_dive/inplace_updates.rst:313]: done

File: docs/overview/deep_dive/inplace_updates.rst (hunk 3 above)
Comment (paulaehab) [reply docs/overview/deep_dive/inplace_updates.rst:335]: done

File: docs/overview/deep_dive/inplace_updates.rst (hunk 4 above)
Comment (paulaehab) [reply docs/overview/deep_dive/inplace_updates.rst:363]: Done

File: docs/overview/deep_dive/inplace_updates.rst (hunk 5 above)
Comment (paulaehab) [reply docs/overview/deep_dive/inplace_updates.rst:414]: Done

File: docs/overview/deep_dive/inplace_updates.rst (hunk 6 above)
Comment (paulaehab) [reply docs/overview/deep_dive/inplace_updates.rst:426]: I didn't get what you want me to do in this section ???

Comment (paulaehab) [reply docs/overview/deep_dive/inplace_updates.rst:426]: Also I think here as the code changed we need to edit the text right ?
```suggestion
Firstly, the return of :func:`ivy.sum` is the same shape and type as the return of the entire function, and so we can also write this output to the :code:`out` argument inplace.
```

File: docs/overview/deep_dive/inplace_updates.rst (hunk 1 above)
Comment (paulaehab) [reply docs/overview/deep_dive/inplace_updates.rst:301]: done

**Review by VedPatwardhan — APPROVED** [review]
lgtm! Feel free to merge the PR, thanks for making the changes @paulaehab 🙂
